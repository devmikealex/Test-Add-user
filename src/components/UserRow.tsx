import { deleteUser } from '../features/usersSlice'
import User from '../types/User'
import { useDispatch } from 'react-redux'
import xIcon from '/assets/x-icon.svg'


interface Props {
    user: User
}

function UserRow({ user }: Props) {
    const dispatch = useDispatch()

    return (
        <tr>
            <td><div className='id-cell'>{user.id}</div></td>
            <td className='clickable' onClick={() => dispatch(deleteUser(user.id))}>
                {user.name}
                <div className='warnIcon'><img src={xIcon} className="x-icon" alt="x icon" /></div>
            </td>
            <td>{user.username}</td>
            <td>{user.email}</td>
        </tr>
    )
}

export default UserRow
