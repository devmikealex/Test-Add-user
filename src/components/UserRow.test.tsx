import { render, screen } from '@testing-library/react'

import { Provider } from 'react-redux'
import { store } from '../store/store'

import UserRow from './UserRow'

const testUser = {
    id: 555,
    name: 'Default Name 555',
    username: 'Test555',
    email: '555@com.com',
}

it('Test row for users props', () => {
    // const table = document.createElement('table')

    const { baseElement, container } = render(
        <Provider store={store}>
            <UserRow user={testUser} />
        </Provider>,
        // { container: table }
        // { baseElement: null }
    )
    screen.debug()
    // expect(baseElement).toMatchSnapshot()
    expect(container).toMatchSnapshot()
    expect(screen.getByText('Default Name 555')).toBeInTheDocument()
    expect(screen.getByText('Test555')).toBeInTheDocument()
    expect(screen.getByText('555@com.com')).toBeInTheDocument()
})
