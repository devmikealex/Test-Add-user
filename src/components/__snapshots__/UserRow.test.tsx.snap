// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test row for users props 1`] = `
<div>
  <tr>
    <td>
      <div
        class="id-cell"
      >
        555
      </div>
    </td>
    <td
      class="clickable"
    >
      Default Name 555
      <div
        class="warnIcon"
      >
        <img
          alt="x icon"
          class="x-icon"
          src="/assets/x-icon.svg"
        />
      </div>
    </td>
    <td>
      Test555
    </td>
    <td>
      555@com.com
    </td>
  </tr>
</div>
`;
